pkg/asset/ibi/ibi_config.go

```diff
@@ -0,0 +1,131 @@
+package ibi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types/ibi"
+)
+
+var ibiConfigFilename = "ibi-config.yaml"
+
+// ImageBasedInstallConfig reads the ibi-config.yaml file.
+type ImageBasedInstallConfig struct {
+	File     *asset.File
+	Config   *ibi.Config
+	Template string
+}
+
+var _ asset.WritableAsset = (*ImageBasedInstallConfig)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*ImageBasedInstallConfig) Name() string {
+	return "Image-based Install Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ImageBasedInstallConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the Image-based Install Config manifest.
+func (i *ImageBasedInstallConfig) Generate(dependencies asset.Parents) error {
+	ibiConfigTemplate := `#
+# Note: This is a sample ImageBasedInstallConfig file showing
+# which fields are available to aid you in creating your
+# own ibi-config.yaml file.
+#
+apiVersion: v1beta1
+kind: ImageBasedInstallConfig
+metadata:
+  name: example-ibi-config
+  namespace: cluster0
+seedImage: quay.io/openshift-kni/seed-image:4.16.0
```

Comment:
The separate tool is a pod that push an OCI image to a registry.
Adding this functionality to that pod will impose requirements such as route and persistent storage that the pod currently doesn't have, in addition, it will need to implement the ISO building logic + serve REST for users to download  this ISO.
On the other hand, the installer already implements very similar functionality (creating a live-iso installation image) for the agent-based-installer.
At the end of the day all this is doing is:
1. Generate ignition config according to user input
2. Downloading the rhocs live image 
3. Embed the ignition to the live image

Reply:
We should probably have this discussion on the enhancement instead of here, but there is definitely no need for persistent storage, you'd just use machine-os-images to as an init container to extract the base ISO to temporary local storage. Implementing an HTTP server in a golang program is trivial. The ISO building logic already exists in the assisted-image-service library (in fact, that's where the installer consumes it from anyway). We do exactly the same stuff already in the image-customization-controller that builds ISOs for baremetal-operator to boot.